lib/dry/validation/schema.rb

```diff
@@ -92,6 +94,14 @@ def [](name)
       def predicates
         self.class.predicates
       end
+
+      private
+
+      def validate_input(input)
+        unless input.is_a?(Hash)
+          raise ArgumentError, "expected a Hash but received +#{input.inspect}+"
+        end
```

Comment:
@solnic is there a plan to have a similar implementation for attribute based validations? Something like:

```ruby
class Schema < Dry::Validation::Schema
  entity? do
    attr(:name, &:filled?)
  end
end
```

I doubt `entity?` is the right name, but just curious. If that was in place we could potentially use that precondition on the predicate to determine if the input is of the correct type.